src/checker.ml

```diff
@@ -791,3 +791,33 @@ let view_is_burrow_overburrowed (burrow_id, state: burrow_id * checker) : bool =
 
 let view_is_burrow_liquidatable (burrow_id, state: burrow_id * checker) : bool =
   burrow_is_liquidatable state.parameters (find_burrow state.burrows burrow_id)
+
+(* ************************************************************************* *)
+(**                            FA2_VIEWS                                     *)
+(* ************************************************************************* *)
+
+let view_get_balance ((owner, token_id), state: (Ligo.address * fa2_token_id) * checker) : Ligo.nat =
+  fa2_get_balance (state.fa2_state, owner, token_id)
+
+let view_total_supply (token_id, state: fa2_token_id * checker) : Ligo.nat =
+  (* TODO: we should have assertions that the total amounts in parameters and cfmm is consistent with
+     what's in the ledger. Alternatively, we can use the ledger as a source of truth and remove
+     parameters.circulating_kit and cfmm.lqt.
+   *)
```

Comment:
Yeah, that'd be good. I guess then we need to modify ledger to store the total amount per token, and update all other functions to use those. Can you create an issue about it, since I am not 100% sure about the intricacies of `parameters.circulating_kit`?

Reply:
Yeah, I will open an issue and try to address this one, to make sure it agrees with the recent changes to the circulating_kit. :+1: